Views

zipnikitin opened the pull request:
1. Реализуйте все стандартные действия вместе с представлениями (видами) для ресурса вопросов:
Список вопросов теста должен выводиться на той же странице, что и сам тест - views/tests и views/questions
Используйте фрагменты представлений там, где это уместно - все представления questions и tests разделены на фрагменты и ссылаются друг на друга
Для того чтобы метод form_with смог сформировать верный URL используйте вызов form_with model: [@parent, @child] - использовано для questions/_form и tests/_form
В представлениях должны быть ссылки для навигации между страницами приложения - link_to для questions и tests
2. Добавьте футер в котором будет выводиться текущий год, имя автора, название проекта (в виде ссылки на GitHub-репозиторий проекта), его краткое описание и ссылка с текстом "Учебный проект в онлайн-школе Thinknetica", ведущая на адрес thinknetica.com - shared/footer используется в теле layouts/applications
3. Напишите хэлперы:
Для заголовка формы создания и редактирования вопроса. Это должен быть один метод question_header который в форме создания вопроса будет выводить Create New <Test Title> Question, а в форме редактирования Edit <Test Title> Question
Для вывода текущего года в футере
Для формирования ссылки на GitHub репозиторий: github_url(author, repo) - helpers/questions_helper используется в shared/footer
4. Добавьте в представление списка тестов количество вопросов для каждого теста - test.questions.count в tests._test

zipnikitin committed:
Views

1. Реализуйте все стандартные действия вместе с представлениями (видами) для ресурса вопросов:
Список вопросов теста должен выводиться на той же странице, что и сам тест - views/tests и views/questions
Используйте фрагменты представлений там, где это уместно - все представления questions и tests разделены на фрагменты и ссылаются друг на друга
Для того чтобы метод form_with смог сформировать верный URL используйте вызов form_with model: [@parent, @child] - использовано для questions/_form и tests/_form
В представлениях должны быть ссылки для навигации между страницами приложения - link_to для questions и tests
2. Добавьте футер в котором будет выводиться текущий год, имя автора, название проекта (в виде ссылки на GitHub-репозиторий проекта), его краткое описание и ссылка с текстом "Учебный проект в онлайн-школе Thinknetica", ведущая на адрес thinknetica.com - shared/footer используется в теле layouts/applications
3. Напишите хэлперы:
Для заголовка формы создания и редактирования вопроса. Это должен быть один метод question_header который в форме создания вопроса будет выводить Create New <Test Title> Question, а в форме редактирования Edit <Test Title> Question
Для вывода текущего года в футере
Для формирования ссылки на GitHub репозиторий: github_url(author, repo) - helpers/questions_helper используется в shared/footer
4. Добавьте в представление списка тестов количество вопросов для каждого теста - test.questions.count в tests._test

VidgarVii commented on testGuru/app/controllers/questions_controller.rb:
```diff
@@ -33,6 +33,18 @@ def create
     end
   end
 
```
важно:
ТЗ: "Список вопросов теста должен выводиться **на той же странице, что и сам тест**"
другими словами нужна страница для вывода теста
что приводит к следующим выводам:
- экшен `index` в классе `QuestionsController` лишний
- вызов коллбека для `index` не нужен
- список вопросов нужно выводить на странице теста
- путь `test_questions_path`, который ты используешь в экшенах, тоже лишний (нужно отменить его создание в `routes`)
- нужен контроллер для ресурса `:tests` (его написание было поэтапно продемонстрировано в скринкасте)

VidgarVii commented on testGuru/app/controllers/questions_controller.rb:
```diff
@@ -33,6 +33,18 @@ def create
+  end
+
+  def update
+    if @question.update(question_params)
```
тут ошибка. протестируй экшен

zipnikitin committed:
Views

экшен index в классе QuestionsController лишний
вызов коллбека для index не нужен - починил QuestionsController, убрал index, поправил фильтры, подогнал под TestsController
список вопросов нужно выводить на странице теста - таблица в tests/show
путь test_questions_path, который ты используешь в экшенах, тоже лишний (нужно отменить его создание в routes) - сделал исключение для index в routes
нужен контроллер для ресурса :tests (его написание было поэтапно продемонстрировано в скринкасте) - пересобрал контроллер тестов согласно скринкасту
Также:
Поправил отображение ссылок, теперь они в нужных местах
Починил отображение года
Добавил shared/errors согласно скринкасту
Вставил хедеры везде где нужно
Сделал подтверждение удаления согласно скринкасту
Пересобрал форму теста согласно скринкасту, по её образу пересобрал форму вопроса
Собрал индекс согласно скринкасту

VidgarVii commented on testGuru/app/helpers/questions_helper.rb:
```diff
@@ -0,0 +1,19 @@
+module QuestionsHelper
+  def question_header(question)
+    @test_title = question.test.tittle
```
тут нет смыcла использовать инстанс переменную. Она используется только внутри метода

VidgarVii commented on testGuru/app/helpers/questions_helper.rb:
```diff
@@ -0,0 +1,19 @@
+  end
+
+  def current_year
+    Date.today.year
```
в рельсах рекомендуется использовать `Time.current` вместо `Time.now` (`Time.new`) и `Date.current` вместо `Date.today`
`.current` возвращает текущий момент времени с учетом настройки часового пояса в приложении

zipnikitin committed:
Views

Скорректировал время и сделал локальную переменную вместо инстанс test_title.

VidgarVii commented on .gitignore:
```diff
@@ -8,3 +8,7 @@ testGuru/db/.DS_Store
+testGuru/app/views/.DS_Store
+testGuru/app/views/.DS_Store
+.DS_Store
+testGuru/app/views/.DS_Store
```
обычно 1 раза хватает
